### PR TITLE
Add `SeccompDefault` kubelet config

### PR DESCRIPTION
**Pull request by colinhoglund**
Updating the kubelet config spec with the `SeccompDefault` feature gate and `--seccomp-default` boolean flag.

### CODE REVIEW

File: pkg/apis/kops/validation/validation.go [hunk 1]
```diff
@@ -922,6 +922,10 @@ func validateKubelet(k *kops.KubeletConfigSpec, c *kops.Cluster, kubeletPath *fi
 		if k.MemorySwapBehavior != "" {
 			allErrs = append(allErrs, IsValidValue(kubeletPath.Child("memorySwapBehavior"), &k.MemorySwapBehavior, []string{"LimitedSwap", "UnlimitedSwap"})...)
 		}
+
+		if k.SeccompDefault != nil && c.IsKubernetesLT("1.22") {
+			allErrs = append(allErrs, field.Forbidden(kubeletPath.Child("seccompDefault"), "seccompDefault is only available in Kubernetes >=1.22"))
+		}
```
Comment (hakman) [inline pkg/apis/kops/validation/validation.go:928]: Please remove the validation. K8s 1.22 is no longer supported for some time now.

**Review by hakman — CHANGES_REQUESTED** [review]

File: docs/cluster_spec.md [hunk 2]
```diff
@@ -816,6 +816,20 @@ spec:
...
+[SeccompDefault](https://kubernetes.io/blog/2021/08/25/seccomp-default/) enables the use of `RuntimeDefault` as the default seccomp profile for all workloads. (Default: false)
+
+Note that a feature gate is required to enable the feature, and the feature is turned on using kubelet config.
+
+```yaml
+spec:
+  kubelet:
+    featureGates:
+      SeccompDefault: "true"
+    seccompDefault: true
```
Comment (colinhoglund) [inline docs/cluster_spec.md:830]: This feels a bit unnecessary. Would it make sense to just automatically enable the feature flag when `seccompDefault: true`?

Comment (colinhoglund) [reply docs/cluster_spec.md]: I went ahead and pushed a commit for this change. Happy to revert if there is a preferred way to handle this type of configuration.

Comment (hakman) [reply docs/cluster_spec.md]: Thanks for the thought, but it is not the direction we want to go. We expect operators to set these flags and maintain them.

File: nodeup/pkg/model/kubelet_test.go [hunk 3]
```diff
@@ -411,3 +412,24 @@ func Test_BuildComponentConfigFile(t *testing.T) {
 		t.Errorf("Failed to build component config file: %v", err)
 	}
 }
+
+func TestSeccompDefault(t *testing.T) {
+	b := &KubeletBuilder{
+		&NodeupModelContext{
+			NodeupConfig: &nodeup.Config{
+				KubeletConfig: kops.KubeletConfigSpec{
+					SeccompDefault: ptr.To(true),
```
Comment (hakman) [inline nodeup/pkg/model/kubelet_test.go:421]:
```suggestion
					SeccompDefault: fi.PtrTo(true),
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: nodeup/pkg/model/kubelet_test.go (hunk 3 above)
Comment (hakman) [reply nodeup/pkg/model/kubelet_test.go:421]: Once you do these changes, please also squash the commits.

**Comment by hakman** [comment]
Thanks @colinhoglund! 🚀 
/lgtm
/approve

**Pull request merged**